app/scripts/lib/transaction/sentinel-api.ts

```diff
@@ -0,0 +1,59 @@
+import { Hex } from '@metamask/utils';
+import { hexToDecimal } from '../../../../shared/modules/conversion.utils';
+import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
+
+const BASE_URL = 'https://tx-sentinel-{0}.api.cx.metamask.io/';
+const ENDPOINT_NETWORKS = 'networks';
+
+export type SentinelNetwork = {
+  name: string;
+  group: string;
+  chainID: number;
+  nativeCurrency: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+  network: string;
+  explorer: string;
+  confirmations: boolean;
+  smartTransactions: boolean;
+  relayTransactions: boolean;
+  hidden: boolean;
+  sendBundle: boolean;
+};
+
+export type SentinelNetworkMap = Record<string, SentinelNetwork>;
+
+/**
+ * Returns all network data.
+ */
+export async function getNetworkData(): Promise<SentinelNetworkMap> {
```

Comment:
Minor, can we name this something with more context such as `getAllSentinelNetworkFlags`?

Also why does this have to be exported if the callers will only ever ask about specific flags?

Reply:
Renamed and removed the export.